[Khoo Wui Hong] IP

wui-hong opened the pull request:
# Hong

Your personal [assistant](https://dictionary.cambridge.org/dictionary/english/assistant)
Hong can do the following for you:
- Track tasks
- Set deadlines

All you have to do is:
1. **Download** it
2. Leave a good review (optional)

- [x] Update tasks
- [x] Update more tasks
- [ ] Update even MORE tasks 👍 


If you are a Java programmer, you can use it to practice Java too. Here's the a sneak peek into the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

wui-hong committed:
Complete Level 0

wui-hong committed:
Complete Level 1 Echo

wui-hong committed:
Complete Level 2 List

wui-hong committed:
Complete Level 3 Mark as Done

wui-hong committed:
Complete Level 4

wui-hong committed:
Complete A-TextUiTesting

wui-hong committed:
Complete Level 5

wui-hong committed:
Complete Level-6

wui-hong committed:
Complete Level-7

wui-hong committed:
Merge branch 'branch-Level-7'

wui-hong committed:
Update Deadline, Event, Task, Todo, DateTimeParser to have DateTime
variables. Left with implementing changes in Hong.java

wui-hong committed:
Update createDeadline function

wui-hong committed:
Update createEvent function

wui-hong committed:
Complete Level-8

wui-hong committed:
Merge branch 'branch-Level-8'

wui-hong committed:
Complete Ui portion of A-MoreOOP

wui-hong committed:
Complete Storage of A-MoreOOP

wui-hong committed:
Complete TaskList of A-MoreOOP

wui-hong committed:
Complete Parser of A-MoreOOP

wui-hong committed:
Complete A-Packages

wui-hong committed:
Merge branch 'add-gradle-support'

wui-hong committed:
Complete A-JUnit

wui-hong committed:
Complete A-Jar

wui-hong committed:
Complete A-Jar

wui-hong committed:
Revert "Complete A-Jar"

This reverts commit ed222c6a4bed779db9377e6061ac6d474afeddb3.

wui-hong committed:
Touch up some code

wui-hong committed:
Touch up more code

wui-hong committed:
Update Storage to create file paths

wui-hong committed:
Complete A-JavaDoc, A-CodingStandard, Level-9

Gavino3o commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+        this.tasks = Storage.readTasks();
+    }
+
+    public void printTasks() {
```
Consistent use of camelCase and verbs to describe methods. LGTM. 👍

Gavino3o commented on src/main/java/hong/Hong.java:
```diff
@@ -0,0 +1,69 @@
+     * @param args The command line arguments.
+     */
+    public static void main(String[] args) {
+        boolean isLoopContinuing = true;
```
Can consider:
```suggestion
        boolean isActive = true;
```

Gavino3o commented on src/main/java/parsers/DateTimeParser.java:
```diff
@@ -0,0 +1,31 @@
+     * @return The LocalDateTime object constructed from the String.
+     */
+    public static LocalDateTime readTasksParser(String dateTime) {
+        //2019-10-15 1800
```
Consider to remove this comment in the future.

Gavino3o commented on src/main/java/hong/Hong.java:
```diff
@@ -0,0 +1,69 @@
+import tasks.TaskList;
+import ui.Ui;
+
+
```
Consider adding a JavaDoc here as it is an important class for this application.

Gavino3o reviewed (commented):
All in all looks good to me 👍 ! But do consider adding JavaDocs to each your classes as it is required in the coding standard provided. Cheers and have a good day.

qyaner commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,106 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
```
maybe can add javadocs for classes and not just methods to show the general function of the class

qyaner commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,106 @@
+package storage;
+
+import parsers.DateTimeParser;
+import tasks.Task;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Todo;
+import ui.Ui;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
```
good job on importing all the packages individually!

qyaner commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+        Ui.print(message);
+        Ui.printLine();;
+    }
+    public void deleteTask(String userInput) {
```
naming generally well done, very clear and concise

qyaner commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+package tasks;
+
+import parsers.DateTimeParser;
+import storage.Storage;
+import ui.Ui;
+import exceptions.DukeException;
```
good job on using multiple package to store the classes! makes the code looks more organized!

qyaner reviewed (commented):
LGTM! In general, other than a few nits here and there, you did a great job in following the coding standards!  👍

wui-hong committed:
Add checkStyle files

wui-hong committed:
Merge branch 'branch-A-CheckStyle'

mamayuan commented on src/main/java/hong/Hong.java:
```diff
@@ -0,0 +1,69 @@
+            String userInput = myObj.nextLine();
+            Parser parser = new Parser();
+            String parsedCommand = parser.parseCommand(userInput);
+            switch (parsedCommand) {
```
I like your structuring of code! It is clear for reading and maybe you can consider adding in more empty lines to separate code chunks. Just to make it easier for reading :)

mamayuan commented on src/main/java/parsers/DateTimeParser.java:
```diff
@@ -0,0 +1,31 @@
+     */
+    public static LocalDateTime parseDateTime(String dateTime) {
+        //2019-10-15 1800
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
```
This is not regarding code standard but I'd like to share what I did, i think i can be useful for you to consider. I have incorporated a list of all acceptable formats and checks whether the input aligns with any of the options. This makes the input format more flexible. Actually can ask GPT to list out for you the possible formats for datetime and date :)

mamayuan commented on src/main/java/parsers/Parser.java:
```diff
@@ -0,0 +1,32 @@
+     * @params userInput The input from the user containing the command.
+     * @return A String containing just the command.
+     */
+    public String parseCommand(String userInput) {
```
Would you consider removing the whitespaces around the input just in case it is accidentally keyed in

mamayuan commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,106 @@
+     * @return An ArrayList<Task>.
+     */
+    public static ArrayList<Task> readTasks() {
+        ArrayList<Task> tasks = new ArrayList<Task>();
```
remember to change to TaskList object later~

mamayuan commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,23 @@
+package ui;
```
remember to add javadoc for this class as well~

mamayuan reviewed (commented):
I like what you are doing. Your variable naming is informative and your code structure is clear! Learning from you as I read :)

wui-hong committed:
Complete Level-10

wui-hong committed:
Merge branch 'branch-Level-10'

wui-hong committed:
No assertions are done for the return value of Parser's parseCommand
function

There could be a possibility that it returns an empty String

Lets create an assertion for the String returned inside Hong.java

It is the only place in the code where parseCommand is called

wui-hong committed:
No assertions are done for the return value of getResponse function

There could be a possibility that it returns an empty String, resulting
in the bot not saying anything to the user

Lets create an assertion for the String returned inside Hong.java
getResponse function

wui-hong committed:
Merge pull request #1 from wui-hong/branch-A-Assertions

Complete A-Assertions

wui-hong committed:
Refactoring of code to delete redundant code that was for the text
based cli, as well as deleting commented out code in Hong, Main, Ui java
files.

wui-hong committed:
Merge pull request #2 from wui-hong/Branch-A-CodeQuality

Merge Branch-A-CodeQuality

wui-hong committed:
Complete BDC-Extension with B-Snooze

Snoozing a deadline will increase the deadline by 1 day, while
snoozing an event will increase its end date by 1 day

wui-hong committed:
Complete A-UserGuide

wui-hong committed:
Update Ui image

wui-hong committed:
Update Ui image again

wui-hong committed:
Refactor code

wui-hong committed:
Update ReadMe

wui-hong committed:
Update ReadMe

wui-hong committed:
Update code for refactoring

wui-hong committed:
Update Storage.java

wui-hong committed:
Add DeadlineTest.java

wui-hong committed:
Remove TaskListTest.java

wui-hong committed:
Update testing
